pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter_test.go

```diff
@@ -151,24 +152,59 @@ var _ = Describe("Validating VM Admitter", func() {
 		virtClient.EXPECT().AuthorizationV1().Return(k8sClient.AuthorizationV1()).AnyTimes()
 	})
 
-	It("reject invalid VirtualMachineInstance spec", func() {
-		vmi := api.NewMinimalVMI("testvmi")
-		vmi.Spec.Domain.Devices.Disks = append(vmi.Spec.Domain.Devices.Disks, v1.Disk{
-			Name: "testdisk",
+	Context("with an invalid VM", func() {
+		It("should reject the request with unrecognized field", func() {
```

Comment:
@vladikr I did not change any behavior here.  I just converted the e2e test to unit test :)
So yes, we were rejecting unrecognized fields even before.
It is something that we do for every resource and looks like we have always done it:
https://github.com/kubevirt/kubevirt/commit/93eecf349031477b2bedbc0bbcbd29407deda82d#diff-37e7f54f2a03b510af54f5d608e432bb0447d09ea6066a9970e0be8beab66649R1214

AFAIK k8s  does not allow unrecognized fields as well.

Also, maybe we can perform this instructing openapi, instead of manually doing this check.

Reply:
Yes, the fields were rejected as previously the Kube would only ignore the fields. In case of typo the behavior could be different..